stubs/grpcio/METADATA.toml

```diff
@@ -0,0 +1,15 @@
+version = "1.*"
+upstream_repository = "https://github.com/grpc/grpc"
+partial_stub = true
+requires = [
+    "types-protobuf",
+]
+
+[tool.stubtest]
+ignore_missing_stub = true
+stubtest_requirements = [
+    "grpcio-channelz",
+    "grpcio-health-checking",
+    "grpcio-reflection",
+    "grpcio-status",
```

Comment:
I just realized, but aren't your top-level packages actually separate distributions?
Shouldn't these all be their own stubs rather than all lumped in `stubs/grpcio` ?

If someone installs `grcpio`, it doesn't come with, for example, the package `grpc_health`. You have to specifically install `grpcio-health-checking`

Reply:
will we need to add `types-grpcio` to `typeshed-internal/stub_uploader` before we can split in that way?

Reply:
Stub-uploader automatically accepts stubs depending on other typeshed stubs IIRC.

But as I mentioned, I don't mind doing that as a follow-up, and seeing if we hit any roadblocks there. Just so this PR doesn't get dragged too much further.